### PR TITLE
Use a cluster by default

### PR DESCRIPTION
**Pull request by GGP1**
|Related issue|
|---|
| Closes https://github.com/wazuh/wazuh/issues/25895 |

## Description

Introduces modifications to always use a cluster of nodes and to set the default bind address to `localhost` (previously `0.0.0.0`).

## Tests

<details><summary>Cluster configuration</summary>

```xml
<cluster>
  <name>wazuh</name>
  <node_name>master-node</node_name>
  <node_type>master</node_type>
  <bind_addr>0.0.0.0</bind_addr>
  <port>1516</port>
  <nodes>
      <node>wazuh-master</node>
  </nodes>
  <hidden>no</hidden>
</cluster>
```

</details>

<details><summary>Master logs</summary>

```console
root@wazuh-master:/# /var/ossec/bin/wazuh-clusterd -f
Starting cluster in foreground (pid: 2007)
2024/09/25 14:41:10 INFO: [Master] [Main] Generating JWT signing key pair
2024/09/25 14:41:10 INFO: [Local Server] [Main] Serving on /var/ossec/queue/cluster/c-internal.sock
2024/09/25 14:41:10 INFO: [Master] [Main] Serving on ('0.0.0.0', 1516)
2024/09/25 14:41:10 INFO: [Master] [Local integrity] Starting.
2024/09/25 14:41:10 INFO: [Master] [Local integrity] Finished in 0.101s. Calculated metadata of 3 files.
2024/09/25 14:41:18 INFO: [Master] [Local integrity] Starting.
2024/09/25 14:41:18 INFO: [Master] [Local integrity] Finished in 0.001s. Calculated metadata of 3 files.
2024/09/25 14:41:26 INFO: [Master] [Local integrity] Starting.
2024/09/25 14:41:26 INFO: [Master] [Local integrity] Finished in 0.001s. Calculated metadata of 3 files.
2024/09/25 14:41:34 INFO: [Master] [Local integrity] Starting.
2024/09/25 14:41:34 INFO: [Master] [Local integrity] Finished in 0.002s. Calculated metadata of 3 files.
2024/09/25 14:41:42 INFO: [Master] [Local integrity] Starting.
2024/09/25 14:41:42 INFO: [Master] [Local integrity] Finished in 0.001s. Calculated metadata of 3 files.
2024/09/25 14:41:50 INFO: [Worker] [Main] Connection from ('172.18.0.5', 52332)
2024/09/25 14:41:50 INFO: [Master] [Local integrity] Starting.
2024/09/25 14:41:50 INFO: [Master] [Local integrity] Finished in 0.002s. Calculated metadata of 3 files.
2024/09/25 14:41:59 INFO: [Master] [Local integrity] Starting.
2024/09/25 14:41:59 INFO: [Master] [Local integrity] Finished in 0.002s. Calculated metadata of 3 files.
2024/09/25 14:41:59 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:41:59 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.002s. Received metadata of 1 files. Sync required.
2024/09/25 14:41:59 INFO: [Worker wazuh-worker1] [Integrity sync] Starting.
2024/09/25 14:41:59 INFO: [Worker wazuh-worker1] [Integrity sync] Files to create in worker: 2 | Files to update in worker: 0 | Files to delete in worker: 0
2024/09/25 14:41:59 INFO: [Worker wazuh-worker1] [Integrity sync] Finished in 0.003s.
2024/09/25 14:42:07 INFO: [Master] [Local integrity] Starting.
2024/09/25 14:42:07 INFO: [Master] [Local integrity] Finished in 0.001s. Calculated metadata of 3 files.
2024/09/25 14:42:09 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:42:09 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.002s. Received metadata of 3 files. Sync not required.
2024/09/25 14:42:15 INFO: [Master] [Local integrity] Starting.
2024/09/25 14:42:15 INFO: [Master] [Local integrity] Finished in 0.001s. Calculated metadata of 3 files.
2024/09/25 14:42:18 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:42:18 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.002s. Received metadata of 3 files. Sync not required.
2024/09/25 14:42:23 INFO: [Master] [Local integrity] Starting.
2024/09/25 14:42:23 INFO: [Master] [Local integrity] Finished in 0.001s. Calculated metadata of 3 files.
2024/09/25 14:42:27 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:42:27 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.002s. Received metadata of 3 files. Sync not required.
2024/09/25 14:42:31 INFO: [Master] [Local integrity] Starting.
2024/09/25 14:42:31 INFO: [Master] [Local integrity] Finished in 0.001s. Calculated metadata of 3 files.
2024/09/25 14:42:36 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:42:36 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.002s. Received metadata of 3 files. Sync not required.
2024/09/25 14:42:39 INFO: [Master] [Local integrity] Starting.
2024/09/25 14:42:39 INFO: [Master] [Local integrity] Finished in 0.001s. Calculated metadata of 3 files.
```

</details>

<details><summary>Worker1 logs</summary>

```console
root@wazuh-worker1:/# /var/ossec/bin/wazuh-clusterd -f
Starting cluster in foreground (pid: 2332)
2024/09/25 14:41:50 INFO: [Local Server] [Main] Serving on /var/ossec/queue/cluster/c-internal.sock
2024/09/25 14:41:50 INFO: [Worker wazuh-worker1] [Main] Successfully connected to master.
2024/09/25 14:41:59 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:41:59 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.007s. Sync required.
2024/09/25 14:41:59 INFO: [Worker wazuh-worker1] [Integrity sync] Starting.
2024/09/25 14:41:59 INFO: [Worker wazuh-worker1] [Integrity sync] Files to create: 2 | Files to update: 0 | Files to delete: 0
2024/09/25 14:41:59 INFO: [Worker wazuh-worker1] [Integrity sync] Finished in 0.002s.
2024/09/25 14:42:08 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:42:09 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.005s. Sync not required.
2024/09/25 14:42:18 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:42:18 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.005s. Sync not required.
2024/09/25 14:42:27 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:42:27 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.006s. Sync not required.
2024/09/25 14:42:36 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:42:36 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.005s. Sync not required.
2024/09/25 14:42:45 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:42:45 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.014s. Sync not required.
2024/09/25 14:42:50 INFO: [Worker wazuh-worker1] [Keep Alive] Successful response from master: keepalive
2024/09/25 14:42:54 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:42:54 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.004s. Sync not required.
```

</details>

<details><summary>Shutting down the master node</summary>

Master

```console
2024/09/25 14:48:18 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:48:18 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.002s. Received metadata of 3 files. Sync not required.
2024/09/25 14:48:23 INFO: [Master] [Local integrity] Starting.
2024/09/25 14:48:23 INFO: [Master] [Local integrity] Finished in 0.001s. Calculated metadata of 3 files.
2024/09/25 14:48:27 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:48:27 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.002s. Received metadata of 3 files. Sync not required.
2024/09/25 14:48:30 INFO: [Cluster] [Main] SIGINT received. Bye!
```

Worker

```console
2024/09/25 14:47:51 INFO: [Worker wazuh-worker1] [Keep Alive] Successful response from master: keepalive
2024/09/25 14:47:51 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:47:51 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.006s. Sync not required.
2024/09/25 14:48:00 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:48:00 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.005s. Sync not required.
2024/09/25 14:48:09 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:48:09 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.004s. Sync not required.
2024/09/25 14:48:18 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:48:18 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.005s. Sync not required.
2024/09/25 14:48:27 INFO: [Worker wazuh-worker1] [Integrity check] Starting.
2024/09/25 14:48:27 INFO: [Worker wazuh-worker1] [Integrity check] Finished in 0.005s. Sync not required.
2024/09/25 14:48:30 INFO: [Worker wazuh-worker1] [Main] The master closed the connection
```

</details>

<details><summary>Cluster nodes info</summary>

```console
{
    "data": {
        "affected_items": [
            {
                "name": "master-node",
                "type": "master",
                "version": "5.0.0",
                "ip": "wazuh-master"
            },
            {
                "name": "wazuh-worker1",
                "type": "worker",
                "version": "5.0.0",
                "ip": "172.18.0.5"
            }
        ],
        "total_affected_items": 2,
        "total_failed_items": 0,
        "failed_items": []
    },
    "message": "All selected nodes information was returned",
    "error": 0
}
```

</details>

<details><summary>Cluster nodes healthcheck</summary>

```console
{
    "data": {
        "affected_items": [
            {
                "info": {
                    "name": "master-node",
                    "type": "master",
                    "version": "5.0.0",
                    "ip": "wazuh-master",
                    "n_active_agents": 1
                }
            },
            {
                "info": {
                    "name": "wazuh-worker1",
                    "type": "worker",
                    "version": "5.0.0",
                    "ip": "172.18.0.5",
                    "n_active_agents": 0
                },
                "status": {
                    "sync_integrity_free": true,
                    "last_check_integrity": {
                        "date_start_master": "2024-09-25T14:52:00.738823Z",
                        "date_end_master": "2024-09-25T14:52:00.740223Z"
                    },
                    "last_sync_integrity": {
                        "date_start_master": "n/a",
                        "date_end_master": "n/a",
                        "total_extra_valid": 0,
                        "total_files": {
                            "missing": 0,
                            "shared": 0,
                            "extra": 0,
                            "extra_valid": 0
                        }
                    },
                    "last_keep_alive": "2024-09-25T14:51:11.373684Z"
                }
            }
        ],
        "total_affected_items": 2,
        "total_failed_items": 0,
        "failed_items": []
    },
    "message": "All selected nodes healthcheck information was returned",
    "error": 0
}
```

</details>

<details><summary>Cluster status</summary>

```console
{
    "data": {
        "running": "yes"
    },
    "error": 0
}
```

</details>

<details><summary>Master logs without specifying a bind_addr</summary>

```console
2024/09/25 15:17:00 INFO: [Master] [Main] Serving on ('::1', 1516, 0, 0)
2024/09/25 15:17:00 INFO: [Master] [Local integrity] Starting.
2024/09/25 15:17:00 INFO: [Master] [Local integrity] Finished in 0.079s. Calculated metadata of 3 files.
2024/09/25 15:17:08 INFO: [Master] [Local integrity] Starting.
2024/09/25 15:17:10 INFO: [Master] [Local integrity] Finished in 1.063s. Calculated metadata of 3 files.
2024/09/25 15:17:18 INFO: [Master] [Local integrity] Starting.
2024/09/25 15:17:18 INFO: [Master] [Local integrity] Finished in 0.001s. Calculated metadata of 3 files.
2024/09/25 15:17:26 INFO: [Master] [Local integrity] Starting.
2024/09/25 15:17:27 INFO: [Master] [Local integrity] Finished in 0.985s. Calculated metadata of 3 files.
2024/09/25 15:17:35 INFO: [Master] [Local integrity] Starting.
2024/09/25 15:17:35 INFO: [Master] [Local integrity] Finished in 0.380s. Calculated metadata of 3 files.
```

</details>

### CODE REVIEW

**Review by nico-stefani — CHANGES_REQUESTED** [review]
GJ! Just a small thing to note

**Review by nico-stefani — DISMISSED** [review]
LGTM

File: framework/wazuh/manager.py
Comment (fdalmaup) [inline framework/wazuh/manager.py:1]: We should analyze if the `manager:` related actions should be completely removed and how we will manage the permissions related to it.

**Review by fdalmaup — CHANGES_REQUESTED** [review]

File: framework/wazuh/rbac/default/policies.yaml [hunk 1]
```diff
@@ -315,21 +311,12 @@ default_policies:
         effect: allow
 
   cluster_read:
-    description: Provide read access to all cluster/manager related functionalities.
+    description: Provide read access to all cluster related functionalities.
     policies:
-      resourceless:
-        actions:
-          - cluster:status
```
Comment (fdalmaup) [inline framework/wazuh/rbac/default/policies.yaml:322]: We should not remove this.

**Review by fdalmaup — APPROVED** [review]
LGTM! GJ :clap:

**Pull request merged**